[llvm][docs] Add example use of `llvm-reduce`.

fpetrogalli opened the pull request:
The script is based on an example provided by Tim
Northover (https://github.com/TNorthover).

fpetrogalli committed:
[llvm][docs] Add example use of `llvm-reduce`.

The script is based on an example provided by Tim
Northover (https://github.com/TNorthover).

fpetrogalli committed:
Wrap `llvm-reduce` into `:program:`.

arsenm commented on llvm/docs/CommandGuide/llvm-reduce.rst:
```diff
@@ -105,4 +105,33 @@ EXIT STATUS
+
+.. code-block:: bash
+
+   $(LLVM_BUILD_FOLDER)/bin/opt -O2 -S $1 -o /dev/null \
```
Would be good to use -disable-output instead of -o /dev/null

fpetrogalli replied:
Done.

fpetrogalli committed:
Address code review.

1. Disable output of opt instead of dumping output to /dev/null
2. Do not specify -S
